Initial webfact api pull request

sgc-drop42 opened the pull request.

schisnall committed:
Initial commit

schisnall committed:
Initial Commit

schisnall committed:
Initial commit

schisnall committed:
fixed services inc function name

schisnall committed:
update api call work

Commit:
removed var_dump

Commit:
Updated tests, read me,  website field data retrieval

Commit:
added access-control-allow-origin header

Commit:
Modifed website field api return to handle "0" returns

Commit:
added owner_uid to website create call
added url to get call

Commit:
updated website url api return

Commit:
updated tests in module
added call to create website container in create api call

schisnall committed:
modified test domain, changed module package

schisnall committed:
change to default web controller create on create call

schisnall committed:
pass 0 to container create to stop verbose

schisnall committed:
Update read
Update endpoint variables

Boran commented:
Great stuff!
- Should the url not be using webfact as opposed to webfact_api (same as module name, simpler)
- .info: add webfact as a dependancy
- Has this been tested with https too?
- rebase commit into a single initial?

schisnall committed:
Changed api endpoint to 'webfact'
Disabled endpoint debug
Added webfact module dependency

schisnall committed:
Merge branch 'master' of github.com:sgc-drop42/webfactapi

sgc-drop42 commented:
Hello Boran, 
I have modified the service endpoint to 'webfact' 
Added webfact as a dependency 
Haven't tested with the api with https or any form of authentication due to unknown requirements around nova auth.
Tried to rebase to one commit but had some trouble. 

Cheers
Simon

Boran commented:
Rebasing: if stuck, just create a new branch, copy in changes and a new PR.
Merging anyway since this is dev code, but this cannot be used until there is authentication, you cannot expect anyone to enable this module without that.. I suggest at least some key based authentication.
